sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlOptions.scala

```diff
@@ -63,8 +63,10 @@ private[sql] class XmlOptions(
   }
 
   val compressionCodec = parameters.get(COMPRESSION).map(CompressionCodecs.getCodecClassName)
-  val rowTag = parameters.getOrElse(ROW_TAG, XmlOptions.DEFAULT_ROW_TAG)
-  require(rowTag.nonEmpty, s"'$ROW_TAG' option should not be empty string.")
+  val rowTagOpt = parameters.get(ROW_TAG)
+  require(rowTagOpt.isDefined, s"'$ROW_TAG' option is required.")
```

Comment:
While I agree with the logic, I think we should probably add a conf to restore this behaviour back (at `SQLConf`) because we now change the default behaviour. Should also probably add a note at https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md

Otherwise LGTM